tsfc/finatinterface.py

```diff
@@ -31,79 +31,16 @@
-    "Discontinuous Taylor": finat.DiscontinuousTaylor,
-    "Gauss-Legendre": finat.GaussLegendre,
-    "Gauss-Lobatto-Legendre": finat.GaussLobattoLegendre,
-    "HDiv Trace": finat.HDivTrace,
-    "Hellan-Herrmann-Johnson": finat.HellanHerrmannJohnson,
-    "Johnson-Mercier": finat.JohnsonMercier,
-    "Nonconforming Arnold-Winther": finat.ArnoldWintherNC,
-    "Conforming Arnold-Winther": finat.ArnoldWinther,
-    "Hu-Zhang": finat.HuZhang,
-    "Hermite": finat.Hermite,
-    "Kong-Mulder-Veldhuizen": finat.KongMulderVeldhuizen,
-    "Argyris": finat.Argyris,
-    "Hsieh-Clough-Tocher": finat.HsiehCloughTocher,
-    "QuadraticPowellSabin6": finat.QuadraticPowellSabin6,
-    "QuadraticPowellSabin12": finat.QuadraticPowellSabin12,
-    "Reduced-Hsieh-Clough-Tocher": finat.ReducedHsiehCloughTocher,
-    "Mardal-Tai-Winther": finat.MardalTaiWinther,
-    "Alfeld-Sorokina": finat.AlfeldSorokina,
-    "Arnold-Qin": finat.ArnoldQin,
-    "Reduced-Arnold-Qin": finat.ReducedArnoldQin,
-    "Christiansen-Hu": finat.ChristiansenHu,
-    "Guzman-Neilan 1st kind H1": finat.GuzmanNeilanFirstKindH1,
-    "Guzman-Neilan 2nd kind H1": finat.GuzmanNeilanSecondKindH1,
-    "Guzman-Neilan Bubble": finat.GuzmanNeilanBubble,
-    "Guzman-Neilan H1(div)": finat.GuzmanNeilanH1div,
-    "Morley": finat.Morley,
-    "Bell": finat.Bell,
-    "Lagrange": finat.Lagrange,
-    "Nedelec 1st kind H(curl)": finat.Nedelec,
-    "Nedelec 2nd kind H(curl)": finat.NedelecSecondKind,
-    "Raviart-Thomas": finat.RaviartThomas,
-    "Regge": finat.Regge,
-    "Gopalakrishnan-Lederer-Schoberl 1st kind": finat.GopalakrishnanLedererSchoberlFirstKind,
-    "Gopalakrishnan-Lederer-Schoberl 2nd kind": finat.GopalakrishnanLedererSchoberlSecondKind,
-    "BDMCE": finat.BrezziDouglasMariniCubeEdge,
-    "BDMCF": finat.BrezziDouglasMariniCubeFace,
-    # These require special treatment below
-    "DQ": None,
-    "Q": None,
-    "RTCE": None,
-    "RTCF": None,
-    "NCE": None,
-    "NCF": None,
-    "Real": finat.Real,
-    "DPC": finat.DPC,
-    "S": finat.Serendipity,
-    "SminusF": finat.TrimmedSerendipityFace,
-    "SminusDiv": finat.TrimmedSerendipityDiv,
-    "SminusE": finat.TrimmedSerendipityEdge,
-    "SminusCurl": finat.TrimmedSerendipityCurl,
-    "DPC L2": finat.DPC,
-    "Discontinuous Lagrange L2": finat.DiscontinuousLagrange,
-    "Gauss-Legendre L2": finat.GaussLegendre,
-    "DQ L2": None,
-    "Direct Serendipity": finat.DirectSerendipity,
-}
+supported_elements = finat.supported_elements.copy()
 """A :class:`.dict` mapping UFL element family names to their
 FInAT-equivalent constructors.  If the value is ``None``, the UFL
 element is supported, but must be handled specially because it doesn't
 have a direct FInAT equivalent."""
 
+# These require special treatment below
+for tp_family in ("Q", "DQ", "DQ L2", "RTCE", "RTCF", "NCE", "NCF"):
+    supported_elements[tp_family] = None
```

Comment:
This seems like a bit of a code smell. And this does not actually achieve single source of truth since we still have some Firedrake-specific config.

Reply:
Should this special treatment be moved to FIAT?

Reply:
Probably. Why is it needed?

Reply:
The special treatment is for tensor-product elements. I seems like this file could be moved entirely to finat, since it does the ufl to finat conversion, and finat recently became ufl dependant.

Reply:
That seems reasonable